torch/distributed/algorithms/quantization/quantization.py

```diff
@@ -125,6 +125,16 @@ def wrapper(*args, **kwargs):
             for i, t in enumerate(_dequantize_tensor_list(out_tensors, qtype, quant_loss=quant_loss)):
                 tensors[i] = t
 
+        elif (func == dist.all_to_all_single):
```

Comment:
nit: to avoid another if/else condition here when additional functions are supported, consider turning this into a dispatch table that maps collectives to their quantized versions:

```
dispatch = {
dist.all_to_all_single: quantized_a2a_single_handler,
dist.all_gather: quantized_all_gather_handler,
....
}
```